metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -2225,9 +2250,14 @@ private void listReassigningTopic(ListPartitionReassignmentsResponseData respons
     }
 
     // Visible to test.
-    int getTopicEffectiveMinIsr(String topicName) {
+    int getTopicEffectiveMinIsr(String topicName, Function<String, String> getTopicMinIsrConfig) {
         int currentMinIsr = defaultMinIsr;
-        String minIsrConfig = configurationControl.getTopicConfig(topicName, MIN_IN_SYNC_REPLICAS_CONFIG);
+        String minIsrConfig;
+        if (getTopicMinIsrConfig == null) {
```

Comment:
Can we split the method into two signatures instead of having the null?

E.g., `int getTopicEffectiveMinIsr(String topicName)` and `int getTopicEffectiveMinIsr(String topicName, Function<String, String> getTopicMinIsrConfig)`